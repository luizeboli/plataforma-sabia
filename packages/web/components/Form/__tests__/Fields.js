import React from 'react';
import { render, fireEvent, waitFor } from 'test-utils';
import { Form, InputField, TextField, SelectField, SwitchField, CheckBoxField } from '..';

const commonFieldsTests = [
	[`InputField`, InputField],
	[`TextField`, TextField],
	['SelectField', SelectField],
];

test.each(commonFieldsTests)(
	'%s renders properly and on submit shows validation errors',
	async (_, Comp) => {
		const onSubmit = jest.fn(() => {});
		const { container, findByText } = render(
			<Form onSubmit={onSubmit}>
				<Comp
					name="test-input"
					label="Test Input"
					type="text"
					help={<p>Help Text</p>}
					validation={{ required: true }}
				/>
			</Form>,
		);

		fireEvent.submit(container.querySelector('form'));
		await findByText('error:requiredField');
		// if validaton is not succesfull onSubmit must not be called
		expect(onSubmit).not.toHaveBeenCalled();
		expect(container).toMatchSnapshot();
	},
);

test.each(commonFieldsTests)('%s does not trigger validation if not required', async (_, Comp) => {
	const onSubmit = jest.fn(() => {});
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<Comp
				name="test-input"
				label="Test Input"
				type="text"
				validation={{ required: false }}
			/>
		</Form>,
	);

	fireEvent.submit(container.querySelector('form'));
	await waitFor(() => expect(onSubmit).toHaveBeenCalled());

	expect(container).toMatchSnapshot();
});

test('SwitchField works as expected', () => {
	const { container } = render(
		<Form>
			<SwitchField name="test-input" label="Test Input" />
		</Form>,
	);

	expect(container).toMatchSnapshot();
	expect(container.querySelector('label[for=test-input]')).toHaveTextContent('NÃ£o');
	fireEvent.click(container.querySelector('input'));
	expect(container.querySelector('label[for=test-input]')).toHaveTextContent('Sim');
});

test('CheckBoxField works as expected', () => {
	const { container } = render(
		<Form>
			<CheckBoxField name="test-input" label="Test Input" />
		</Form>,
	);

	expect(container).toMatchSnapshot();
});
