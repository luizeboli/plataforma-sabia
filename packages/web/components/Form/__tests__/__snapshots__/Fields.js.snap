// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxField works as expected 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
  font: 1.2rem;
}

.c2 {
  width: 2.5rem;
  height: 2.5rem;
  margin: 0rem;
  margin: 0 0.5rem 0 0;
}

.c3 {
  width: 100%;
  color: hsla(0,0%,40%);
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <input
        aria-label="Test Input"
        aria-required="false"
        class="c2"
        id="test-input"
        name="test-input"
        type="checkbox"
      />
      <div
        class="c3"
      >
        Test Input
      </div>
    </div>
  </form>
</div>
`;

exports[`InputField does not trigger validation if not required 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="test-input"
      >
        Test Input
      </label>
      <div
        class="c3"
      >
        <input
          aria-label="Test Input"
          aria-required="false"
          class="c4"
          id="test-input"
          name="test-input"
          placeholder=""
          type="text"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`InputField renders properly and on submit shows validation errors 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > input {
  border: 1px solid hsla(29,94%,62%);
}

.c1 > .react-select-container .react-select__control {
  border-color: hsla(29,94%,62%);
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c6 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  width: 4rem;
  height: 4rem;
  margin: 0.6rem 0.5rem;
  fill: hsla(0,0%,80%);
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c5:hover {
  cursor: pointer;
  fill: hsla(0,0%,7%);
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="test-input"
      >
        Test Input
      </label>
      <div
        class="c3"
      >
        <input
          aria-label="Test Input"
          aria-required="true"
          class="c4"
          id="test-input"
          name="test-input"
          placeholder=""
          type="text"
        />
        <svg
          class="c5"
          currentItem="false"
          data-for="test-input"
          data-tip="true"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 0 0-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
          />
        </svg>
        <div
          class="__react_component_tooltip uuid place-right type-dark"
          data-id="tooltip"
          id="test-input"
        >
          <style>
            
  	.uuid {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.uuid.place-top {
        margin-top: -10px;
    }
    .uuid.place-top::before {
        border-top: 8px solid transparent;
    }
    .uuid.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .uuid.place-bottom {
        margin-top: 10px;
    }
    .uuid.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .uuid.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .uuid.place-left {
        margin-left: -10px;
    }
    .uuid.place-left::before {
        border-left: 8px solid transparent;
    }
    .uuid.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .uuid.place-right {
        margin-left: 10px;
    }
    .uuid.place-right::before {
        border-right: 8px solid transparent;
    }
    .uuid.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
          </style>
          <p>
            Help Text
          </p>
        </div>
      </div>
      <span
        class="c6"
      >
        error:requiredField
      </span>
    </div>
  </form>
</div>
`;

exports[`SelectField does not trigger validation if not required 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  margin: 1rem 0;
  font-size: 1.4rem;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="test-input"
      >
        Test Input
      </label>
      <div
        class="c3"
      >
        <div
          class="c4 react-select-container css-2b097c-container"
          id="test-input"
        >
          <div
            class="react-select__control css-yk16xz-control"
          >
            <div
              class="react-select__value-container css-g1d714-ValueContainer"
            >
              <div
                class="react-select__placeholder css-1wa3eu0-placeholder"
              >
                Select...
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class="react-select__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    aria-label="Test Input"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-3-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="test-input"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`SelectField renders properly and on submit shows validation errors 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > input {
  border: 1px solid hsla(29,94%,62%);
}

.c1 > .react-select-container .react-select__control {
  border-color: hsla(29,94%,62%);
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c6 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  width: 4rem;
  height: 4rem;
  margin: 0.6rem 0.5rem;
  fill: hsla(0,0%,80%);
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c5:hover {
  cursor: pointer;
  fill: hsla(0,0%,7%);
}

.c4 {
  width: 100%;
  margin: 1rem 0;
  font-size: 1.4rem;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="test-input"
      >
        Test Input
      </label>
      <div
        class="c3"
      >
        <div
          class="c4 react-select-container css-2b097c-container"
          id="test-input"
        >
          <div
            class="react-select__control css-yk16xz-control"
          >
            <div
              class="react-select__value-container css-g1d714-ValueContainer"
            >
              <div
                class="react-select__placeholder css-1wa3eu0-placeholder"
              >
                Select...
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class="react-select__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    aria-label="Test Input"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-2-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="test-input"
            type="hidden"
            value=""
          />
        </div>
        <svg
          class="c5"
          currentItem="false"
          data-for="test-input"
          data-tip="true"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 0 0-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
          />
        </svg>
        <div
          class="__react_component_tooltip uuid place-right type-dark"
          data-id="tooltip"
          id="test-input"
        >
          <style>
            
  	.uuid {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.uuid.place-top {
        margin-top: -10px;
    }
    .uuid.place-top::before {
        border-top: 8px solid transparent;
    }
    .uuid.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .uuid.place-bottom {
        margin-top: 10px;
    }
    .uuid.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .uuid.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .uuid.place-left {
        margin-left: -10px;
    }
    .uuid.place-left::before {
        border-left: 8px solid transparent;
    }
    .uuid.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .uuid.place-right {
        margin-left: 10px;
    }
    .uuid.place-right::before {
        border-right: 8px solid transparent;
    }
    .uuid.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
          </style>
          <p>
            Help Text
          </p>
        </div>
      </div>
      <span
        class="c6"
      >
        error:requiredField
      </span>
    </div>
  </form>
</div>
`;

exports[`SwitchField works as expected 1`] = `
.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c1 {
  margin: 0.5rem 0 1rem 0;
}

.c3 {
  height: 0;
  width: 0;
  visibility: hidden;
}

.c3:checked + label > span {
  left: calc(100% - 2px);
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c4 {
  width: 80px;
  height: 40px;
  margin-top: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  cursor: pointer;
  background: white;
  border-radius: 80px;
  border: 2px solid hsla(204,70%,53%);
  position: relative;
  -webkit-transition: color background-color 0.5s;
  transition: color background-color 0.5s;
}

.c4 > p {
  color: hsla(204,70%,53%);
  padding: 0.5em;
  font-size: 0.8em;
  font-weight: 700;
  text-transform: uppercase;
}

.c4 > span {
  content: '';
  position: absolute;
  top: 1px;
  left: 2px;
  width: calc(40px - 6px);
  height: calc(40px - 6px);
  border-radius: calc(40px - 5px);
  -webkit-transition: 0.2s;
  transition: 0.2s;
  background: hsla(204,70%,53%);
  box-shadow: 0 0 2px 0 rgba(10,10,10,0.29);
}

.c4:active span {
  width: calc(40px + 10px);
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
      >
        Test Input
      </label>
      <input
        class="c3"
        id="test-input"
        name="test-input"
        type="checkbox"
      />
      <label
        class="c4"
        for="test-input"
      >
        <p>
          Não
        </p>
        <span />
      </label>
    </div>
  </form>
</div>
`;

exports[`TextField does not trigger validation if not required 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c5 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 12rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.5rem;
  color: hsla(0,0%,40%);
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="test-input"
      >
        Test Input
      </label>
      <div
        class="c3"
      >
        <textarea
          aria-label="Test Input"
          aria-required="false"
          class="c4"
          id="test-input"
          name="test-input"
          type="text"
        />
      </div>
      <span
        class="c5"
      />
    </div>
  </form>
</div>
`;

exports[`TextField renders properly and on submit shows validation errors 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > input {
  border: 1px solid hsla(29,94%,62%);
}

.c1 > .react-select-container .react-select__control {
  border-color: hsla(29,94%,62%);
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c6 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  width: 4rem;
  height: 4rem;
  margin: 0.6rem 0.5rem;
  fill: hsla(0,0%,80%);
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c5:hover {
  cursor: pointer;
  fill: hsla(0,0%,7%);
}

.c4 {
  width: 100%;
  height: 12rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.5rem;
  color: hsla(0,0%,40%);
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="test-input"
      >
        Test Input
      </label>
      <div
        class="c3"
      >
        <textarea
          aria-label="Test Input"
          aria-required="true"
          class="c4"
          id="test-input"
          name="test-input"
          type="text"
        />
        <svg
          class="c5"
          currentItem="false"
          data-for="test-input"
          data-tip="true"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 0 0-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
          />
        </svg>
        <div
          class="__react_component_tooltip uuid place-right type-dark"
          data-id="tooltip"
          id="test-input"
        >
          <style>
            
  	.uuid {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.uuid.place-top {
        margin-top: -10px;
    }
    .uuid.place-top::before {
        border-top: 8px solid transparent;
    }
    .uuid.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .uuid.place-bottom {
        margin-top: 10px;
    }
    .uuid.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .uuid.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .uuid.place-left {
        margin-left: -10px;
    }
    .uuid.place-left::before {
        border-left: 8px solid transparent;
    }
    .uuid.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .uuid.place-right {
        margin-left: 10px;
    }
    .uuid.place-right::before {
        border-right: 8px solid transparent;
    }
    .uuid.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
          </style>
          <p>
            Help Text
          </p>
        </div>
      </div>
      <span
        class="c6"
      >
        error:requiredField
      </span>
    </div>
  </form>
</div>
`;
